src/main/org/bson/ByteBuf.java

```diff
@@ -488,6 +506,35 @@ public ByteBuf putInt32(ThreadContext context, IRubyObject value) {
     return this;
   }
 
+  /**
+   * Put an unsigned 32 bit integer onto the buffer.
+   *
+   * @param value The integer to write.
+   *
+   * @author Neil Shweky
+   * @since 2020.06.26
+   */
+  @JRubyMethod(name = "put_uint32")
+  public ByteBuf putUInt32(ThreadContext context, IRubyObject value) {
+    if (value instanceof RubyFloat) {
+      value = ((RubyFloat) value).truncate(context);
+    }
+    ensureBsonWrite(4);
+
+    long temp = RubyNumeric.fix2long((RubyFixnum) value);
+    
+    if (temp >= Math.pow(2, 32) || temp < 0)
+      throw getRuntime().newRangeError(format("Number %d is out of range [0, 2^32)", temp));
+    
+    if (temp > Math.pow(2,31)) temp -= Math.pow(2,32);
```

Comment:
Shouldn't this subtract 2**31? If this issue didn't fail any tests can a test be added to detect it?

Reply:
This should be 2^32 actually... Say the number is 4294967295 (2^32-1), this doesn't fit into a signed int, so I have to convert this to it's respective negative number, which is -1. Therefore I have to subtract by 2^32. I can add a test...